### PR TITLE
Some additional partest cleanup

### PR DESCRIPTION
**Pull request by szeiger**
Remove NestUI:
- s.t.p.nest.Colors becomes s.t.p.ConsoleLog
- Move some functionality from NestUI into ConsoleLog
- Move the remainder into AbstractRunner
- Improve TestTranscript formatting
- Remove some unused methods

Make PathSettings immutable

### CODE REVIEW

**Comment by lrytz** [comment]
@som-snytt interested in reviewing this one?

**Comment by som-snytt** [comment]
@lrytz OK, I owe you one or two.

File: src/partest/scala/tools/partest/nest/AbstractRunner.scala [hunk 1]
```diff
@@ -48,9 +40,75 @@ class AbstractRunner(val config: RunnerSpec.Config, protected final val testSour
...
+        }
+        if (logOnFail) showLog()
+      }
+    }
+  }
+
+  def usage(): Unit = {
+    println(RunnerSpec.programInfo.usage)
+    println(RunnerSpec.helpMsg)
+    sys.exit(1)
```
Comment (som-snytt) [inline src/partest/scala/tools/partest/nest/AbstractRunner.scala:102]: could be upgraded to `System.exit(1)`. Adriaan did a sweep last year.

**Review by som-snytt — APPROVED** [review]
Only one nit. Amazing how SZ clears his plate. The next refactor might be better sbt integration? So this is just a stage on the way. For the next few hours, I understand again how the pieces fit. Note to self: use --terse more.

File: src/partest/scala/tools/partest/nest/AbstractRunner.scala (hunk 1 above)
Comment (adriaanm) [reply src/partest/scala/tools/partest/nest/AbstractRunner.scala:102]: will boyscout this Some Day

**Pull request merged**